crates/holochain/src/conductor/conductor.rs

```diff
@@ -2127,41 +2127,61 @@ mod service_impls {
             self.services.share_ref(|s| s.clone())
         }
 
-        pub(crate) async fn initialize_deepkey(
+        pub(crate) async fn initialize_services(self: Arc<Self>) -> ConductorResult<()> {
+            if let Some(installation) = self.get_state().await?.conductor_services.dpki {
+                self.services.share_mut(|s| {
+                    let dpki =
+                        DeepkeyBuiltin::new(self.clone(), self.keystore().clone(), installation);
+                    s.dpki = Some(Arc::new(dpki));
+                });
+            }
+            Ok(())
+        }
+
+        pub(crate) async fn install_dpki(
             self: Arc<Self>,
-            dna: Option<DnaFile>,
+            dna: DnaFile,
+            device_seed_lair_tag: String,
         ) -> ConductorResult<()> {
-            let cell_id = if let Some(dna) = dna {
-                let dna_hash = dna.dna_hash().clone();
-                self.register_dna(dna).await?;
+            let dna_hash = dna.dna_hash().clone();
+            self.register_dna(dna).await?;
 
-                let agent = self.keystore().new_sign_keypair_random().await?;
-                let cell_id = CellId::new(dna_hash, agent);
-                let cell_id_2 = cell_id.clone();
+            let entry = self
+                .keystore()
+                .lair_client()
+                .get_entry(device_seed_lair_tag.clone().into())
+                .await?;
+            let agent = match entry {
+                kitsune_p2p_types::dependencies::lair_keystore_api::lair_store::LairEntryInfo::Seed { tag: _, seed_info } => {
+                    holo_hash::AgentPubKey::from_raw_32(seed_info.ed25519_pub_key.0.to_vec())
+                },
```

Comment:
I still don't understand this. What if the user needs to rekey their dkpi cell? They'd have to invalidate the whole tree down including all the app agent keys, right? Shouldn't we at least introduce another derivation like "device seed -> dpki instance #"?

Reply:
I hadn't considered that fully, but yes that makes sense. I was thinking in terms of devices getting compromised (in which case you have to register a new device), but e.g. key rotation doesn't need to be thought of as a complete change of device. I'll change this.

Reply:
Changed.